### PR TITLE
[#4232] "state" not required for profile page

### LINKED ISSUE #4232: A required state field for user profiles may not make sense for some countries
If you go to the profile page, you'll note that state is a required field.  It may benefit usability to remove the requirement.

https://www.quora.com/Is-there-any-country-without-states

### PR DESCRIPTION
**Pull request by devincowan**
Resolves #4232

### Pull Request Checklist: 
- [x] Positive Test Case Written by Dev

- [ ] Automated Testing

- [ ] Sufficient User and Developer Documentation

- [x] Passing Jenkins Build

- [x] Peer Code review and approval

### Positive Test Case
1. "State" field is updated to "State/Region"
2. State/Region field is no longer required on the signup or profile edit forms
3. There is a dropdown list of states to select from

### CODE REVIEW

**Comment by hydrocheck** [comment]
| METRIC | VALUE |
|:-----:|:-----:|

[https://sonarqube.cuahsi-workstation.com:9000/dashboard?id=hydroshare-4920](https://sonarqube.cuahsi-workstation.com:9000/dashboard?id=hydroshare-4920)

File: theme/forms.py [hunk 1]
```diff
@@ -275,13 +275,11 @@ def clean_user_type(self):
     def clean_country(self):
         data = self.cleaned_data['country']
         if len(data.strip()) == 0:
-            raise forms.ValidationError("Country is a required field.")
+            raise forms.ValidationError("Country/Region is a required field.")
```
Comment (Maurier) [inline theme/forms.py:278]: Shouldn't this validation error list `Country` only since region is now optional?

**Review by Maurier — DISMISSED** [review]
Looks good. Just one minor question.

I also added minor layout improvements.

File: theme/forms.py (hunk 1 above)
Comment (devincowan) [reply theme/forms.py]: you're right thanks, fixed

**Review by Maurier — APPROVED** [review]

**Pull request merged**